src/rules/string-constant-types.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+module.exports = {
+    create(context) {
+        let isFlowAnnotated = false;
+
+        return {
+            Program(node) {
+                isFlowAnnotated = node.comments.some(
+                    (comment) => comment.value.trim() === '@flow'
+                );
+            },
+            ExportNamedDeclaration(node) {
+                if (!isFlowAnnotated) {
+                    return;
+                }
+
+                // E.g. export { default } from './container'
+                if (!node.declaration) {
+                    return;
+                }
+
+                if (node.declaration.kind !== 'const') {
+                    return;
+                }
+
+                const [declaration] = node.declaration.declarations;
+
+                if (!declaration.init || declaration.init.type !== 'Literal') {
+                    return;
+                }
+
+                if (typeof declaration.init.value !== 'string') {
+                    return;
+                }
+
+                if (declaration.id.typeAnnotation) {
+                    return;
+                }
+
+                const maxLiteralLength = 100;
+                if (declaration.init.value.length > maxLiteralLength) {
+                    return;
+                }
+
+                context.report({
+                    node: declaration,
+                    message: 'String literals should be typed',
```

Comment:
I think example in error message would be helpful - after all `export const foo: string = 'a'` is also typed export. IMO it would be better to have more explicit message like `Exported const string literals should be typed, example: export const foo: 'a' = 'a'`

Reply:
Agreed, definitely is too vague at the moment.

I've updated it to show what we're expecting. `export const foo = 'foo'` would now error with

> String literals should be typed. Expected to be of the form `foo: 'foo'`